### PR TITLE
Add ToolTip component

### PR DESCRIPTION
**Pull request by jrichardlai**
Keeping the old API, but I can replace the current tooltip to only have the `ToolTip` component if you prefer @chirag04.

### CODE REVIEW

**Comment by chirag04** [comment]
I definitely like the `ToolTip` component which has a clear mapping between text and actions on them. The old api was a quick hack as it was an urgent requirement for us.

Thoughts on doing a major version bump and keeping the ToolTip component? Any upsides to the old API compared to `ToolTip`?

**Comment by jrichardlai** [comment]
@chirag04 I think we can accomplish most of the use cases with the new `ToolTip` component, but you won't be able to trigger the `ToolTipMenu` programmatically ( which is maybe an edge case ? ).

**Comment by chirag04** [comment]
@jrichardlai  triggering ToolTipMenu programmatically is actually a common case IMO. We should support that.

**Comment by chirag04** [comment]
@jrichardlai So i just played around with `add-tooltip-component` branch. 

About the programmatically trigger, anyone can just set a ref on ToolTip component and call `handleTextPress`. (we should name it better. show maybe?)

**Comment by jrichardlai** [comment]
Okay @chirag04, I can change it, what about `showMenu` ? And so want me to cleanup the code to only have `ToolTip` ?

**Comment by chirag04** [comment]
sure `showMenu` sounds good.

Yeah, can you clean up and only have `ToolTip` component. Also can you add a runnable example. I saw the example in `add-tooltip-component` branch of yours. I used that to play around.

We can change the readme to only show one approach using the `ToolTip` component and add a note about how to programmatically open the menu.

@jrichardlai Amazing work :+1:

**Comment by jrichardlai** [comment]
Thank you for your work!

Ok sounds good, I am planning to make another PR for the example to not pollute the diff :)!
I'll ping you when I'm done with the changes on this PR.

**Comment by jrichardlai** [comment]
Okay it's ready :)

File: README.md [hunk 1]
```diff
@@ -9,23 +9,37 @@ A react-native component from displaying tooltip. Uses UIMenuController.
...
+
+### Example
 
-## Example
 ```javascript
 var React = require('react-native');
 var {
   AppRegistry,
   StyleSheet,
-  Text,
```
Comment (chirag04) [inline README.md:21]: text is needed

File: README.md [hunk 2]
```diff
@@ -33,23 +47,24 @@ var tooltip = React.createClass({
...
-            onChange={this.handleChange}
-            onPress={this.handleFocus}
-            ref={'input'}
+          <ToolTip
+            ref='theToolTip'
+            actions={[
+              {text: 'x', onPress: () => { this.setState({input: 'x pressed'}) }},
+              {text: 'y', onPress: () => { this.setState({input: 'y pressed'}) }}
+            ]}
+            underlayColor={cssVar('trGray10')}
```
Comment (chirag04) [inline README.md]: for demo sake how about 'transparent'?

**Comment by chirag04** [comment]
LGTM. I added you as collab. Feel free to push it

File: README.md (hunk 2 above)
Comment (jrichardlai) [inline README.md]: Oops let me change that.

**Comment by jrichardlai** [comment]
Thank you @chirag04 merging!

**Comment by chirag04** [comment]
@jrichardlai should we do a major version bump now?

**Pull request merged**